File: MedLabO.Models/Requests/MedicinskoOsobljeUpdateRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MedLabO.Models.Requests
{
    public class MedicinskoOsobljeUpdateRequest
    {
        [Required]
        public Guid Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string? Ime { get; set; }

        [Required]
        [MaxLength(70)]
        public string? Prezime { get; set; }

        [Required]
        public bool IsActive { get; set; }

        [Required]
        public int SpolID { get; set; }

        [Required]
        public int ZvanjeID { get; set; }

        [Required]
        [MaxLength(30)]
        public string UserName { get; set; }

        [EmailAddress]
        public string Email { get; set; }

        [Phone]
        public string PhoneNumber { get; set; }
    }
}

File: MedLabO UI/medlabo_desktop/lib/models/uposlenik/medicinsko_osoblje_update_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'medicinsko_osoblje_update_request.g.dart';

@JsonSerializable()
class MedicinskoOsobljeUpdateRequest {
  String? id;
  String? ime;
  String? prezime;
  bool? isActive;
  int? spolID;
  int? zvanjeID;
  String? userName;
  String? email;
  String? phoneNumber;

  MedicinskoOsobljeUpdateRequest(
      {this.ime,
      this.prezime,
      this.isActive,
      this.spolID,
      this.zvanjeID,
      this.userName,
      this.email,
      this.phoneNumber});

  factory MedicinskoOsobljeUpdateRequest.fromJson(Map<String, dynamic> json) =>
      _$MedicinskoOsobljeUpdateRequestFromJson(json);

  Map<String, dynamic> toJson() => _$MedicinskoOsobljeUpdateRequestToJson(this);
}
